var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -188,26 +194,35 @@ def install(self, spec, prefix):
                 '--with-elpa-lib={0}'.format(elpa.libs[0])
             ])
 
-        if '+hdf5' in spec:
+        if self.spec.variants['hdf5'].value != 'none':
```

Comment:
Do we need to take extra care to ensure that `hdf5` is not picked up on the system if found? Like giving `--without-hdf5` or a similar option?

Reply:
@alalazo Good point. Presently the default behavior is not to search for HDF5, since it is not a hard dependency. This might change in the future. So in the absence of `--with-hdf5`, Quantum Espresso does not look for a HDF5. I am working with the Quantum Espresso to fix some of the short comings of the HDF5-hackery (I think we are almost there), but for now we have all this extra ugly logic to deal with.

I have tested all the possible command line arguments:
```
spack install quantum-espresso%gcc@6.2.0 ^mpich ^intel-mkl hdf5=parallel
spack install quantum-espresso%gcc@6.2.0 ^mpich ^intel-mkl hdf5=serial
spack install quantum-espresso%gcc@6.2.0 ^mpich ^intel-mkl hdf5=none
spack install quantum-espresso%gcc@6.2.0 ^mpich ^intel-mkl 
```
And they are build correctly, thus it is safe to merge.